pytorch_lightning/strategies/deepspeed.py

```diff
@@ -321,6 +337,83 @@ def __init__(
         self.hysteresis = hysteresis
         self.min_loss_scale = min_loss_scale
 
+    @property
+    def num_nodes(self) -> int:
+        return self._num_nodes
+
+    @num_nodes.setter
+    def num_nodes(self, num_nodes: int) -> None:
```

Comment:
Seems to be missing `is_distributed` property. This means DistributedSampler won't be applied.